### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by lucaswiman**
@jdavisp3 cc @hrichards This PR updates `baya` so that it is python 3 compatible.

Most of the changes are mechanical transformations from `2to3` and futurize.  I'll annotate the meaningful changes inline.

I also removed the `Makefile`, which is not used anymore, since we're using travis and tox for running tests.

### CODE REVIEW

File: baya/membership.py [hunk 1]
```diff
@@ -35,33 +35,44 @@ def __rxor__(self, other):
 
     # Comparisons.
     def __eq__(self, other):
-        return self.__class__ is other.__class__
+        raise NotImplementedError()
```
Comment (lucaswiman) [inline baya/membership.py:38]: IMO, the old implementation was not something you'd ever want from equality semantics, though it was overridden by child classes in each case.

File: baya/membership.py [hunk 2]
```diff
@@ -174,13 +185,14 @@ def __init__(self, *operands):
                                 operand)
             # TODO: Check if operand is an instance of basestring and cast to
             # a BaseNode?
-        self._operands = operands
+        self._operands = tuple(operands)
+
+    def __hash__(self):
+        return hash((type(self), self._operands))
```
Comment (lucaswiman) [inline baya/membership.py:191]: This assumes that the dynamic roles methods will return the same thing if called repeatedly, which seems like a reasonable assumption for this class to make.

File: baya/membership.py (hunk 1 above)
Comment (jdavisp3) [reply baya/membership.py:38]: 👍

File: baya/membership.py (hunk 2 above)
Comment (jdavisp3) [reply baya/membership.py:191]: Agreed.

**Review by jdavisp3 — APPROVED** [review]
Nice @lucaswiman , LGTM!

**Comment by lucaswiman** [comment]
I'll have as separate PR for django 1.10 compatibility, and then I'll cut a new release. Thanks @jdavisp3!

**Pull request merged**